mrmustard/math/lattice/strategies/vanilla.py

```diff
@@ -188,3 +196,134 @@ def vanilla_vjp(G, c, dLdG) -> tuple[ComplexMatrix, ComplexVector, complex]:  #
     dLdc = np.sum(G_lin.reshape(shape) * dLdG) / c
 
     return dLdA, dLdb, dLdc
+
+
+@njit
+def autoshape_numba(
+    A, b, c, max_prob=0.999, max_shape=100
```

Comment:
Should max_prob be `settings.AUTOCUTOFF_PROBABILITY`?

Reply:
Maybe add a test that changing that setting actually changes the cutoff.

Reply:
This was the behaviour I saw
```
Coherent([0],1).auto_shape() #5
settings.AUTOCUTOFF_PROBABILITY = 0.99999
Coherent([0],1).auto_shape() #still 5
```

Reply:
now it works. note that we're moving away from the word "cutoff" in favour of "shape" so now the setting is "AUTOSHAPE_PROBABILITY"